src/components/wizard/wizard-step.component.ts

```diff
@@ -78,4 +81,22 @@ export class WizardStepComponent {
         this._renderer.setAttribute(this._elementRef.nativeElement, 'id', id);
         this._renderer.setAttribute(this._elementRef.nativeElement, 'aria-labelledby', `${id}-label`);
     }
+
+    setVisited(value: boolean): void {
+        this.visited = value;
+        this.visitedChange.emit(value);
+    }
+
+    setValid(value: boolean): void {
+        if (this._valid === value) {
+            return;
+        }
+
+        const changeVisited = (value && this._valid) || !this._valid;
+        this._valid = value;
+
+        if (changeVisited) {
```

Comment:
I don't think there is a need for this variable here, consider just in-lining the expression